### ISSUE TITLE
Better error messages for non-existent solc

### ISSUE THREAD

**Issue by JSeam2** [zkonduit/ezkl#346]
### Is your proposal related to a problem?

When you run any evm related functions, and solc is missing you get a non-informative final error message ```PanicException: called `Option::unwrap()` on a `None` value```. The log messages do provide hints that solc is missing, but it is still fairly opaque.

```
INFO ezkl_lib.pfsys.srs 2023-07-08 01:43:57,327 srs.rs:23 loading srs from "kzg.params"
INFO ezkl_lib.execute 2023-07-08 01:43:57,341 execute.rs:1561 downsizing params to 16 logrows
INFO ezkl_lib.pfsys 2023-07-08 01:43:57,342 mod.rs:506 loading verification key from "test.vk"
INFO ezkl_lib.graph.model 2023-07-08 01:43:57,343 model.rs:572 configuring model
Traceback (most recent call last):
  File "/opt/homebrew/bin/solc", line 8, in <module>
    sys.exit(solc())
             ^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/solc_select/__main__.py", line 80, in solc
    res = current_version()
          ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/solc_select/solc_select.py", line 62, in current_version
    raise argparse.ArgumentTypeError(
argparse.ArgumentTypeError: No solc version set. Run `solc-select use VERSION` or set SOLC_VERSION environment variable.
thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', /Users/jseam/.cargo/git/checkouts/snark-verifier-6bf5b0ca86356fb5/693e8fb/snark-verifier/src/loader/evm/util.rs:120:61
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
---------------------------------------------------------------------------
PanicException                            Traceback (most recent call last)
Cell In[11], line 10
      6 srs_path = os.path.join('kzg.params')
      7 settings_path = os.path.join('settings.json')
---> 10 res = ezkl_lib.create_evm_verifier(
     11         vk_path,
     12         srs_path,
     13         settings_path,
     14         sol_code_path,
     15         deployment_code_path,
     16     )

PanicException: called `Option::unwrap()` on a `None` value
```

### Describe the solution you'd like

It would be better to have a check for solc in the environment and triggering an explicit error in the evm related functions. Otherwise, the unwrap error messages can be uninformative.

**Comment by gagichce** [comment]
Took a stab at this as I was confused by the same error message! PR up: https://github.com/zkonduit/ezkl/pull/359

**Comment by JSeam2** [comment]
Wow @gagichce thanks for this!

**Comment by gagichce** [comment]
This can be closed as #359 was merged!
